### PR TITLE
Fix working_set calculation in kubelet

### LINKED ISSUE #28619: Eviction manager's observed memory.available is vastly different from /proc/meminfo MemAvailable
I'm working on a node e2e memory eviction test, and even though I set the aggressive hard eviction threshold of ` --eviction-hard memory.available<2.5Gi`, I noticed that my pods were chewing up memory without getting evicted, so I started digging.

I logged the `observations` made from the `synchronize` function in `kubernetes/pkg/kubelet/eviction/eviction_manager.go`, with `glog.Infof("eviction manager: observations: %s", strings.Replace(spew.Sdump(observations), "\n", " ", -1))`, and I saw the following:

`eviction manager: observations: (eviction.signalObservations) (len=1) {  (eviction.Signal) (len=16) "memory.available": (*resource.Quantity)(0xc820d61940)(3377236Ki) }`

sampling `/proc/meminfo` about two seconds later reveals:
```
MemTotal:        3801016 kB
MemFree:          826748 kB
MemAvailable:    1213620 kB
```

Assuming kB and Ki are equivalent units (they seem to both be kibibytes), that's a 2163616Ki difference in reporting between `memory.available` and `MemAvailable`.

My test collectively chews up memory in steps of about 40MB every 10 seconds, so they could show up as different, but shouldn't be THAT different.

Tagging @vishh @derekwaynecarr

### PR DESCRIPTION
**Pull request by vishh**
Bump cadvisor dependencies to latest head.

Fixes #28619

### CODE REVIEW

**Comment by vishh** [comment]
cc @timstclair

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -2160,6 +2160,481 @@
 			"ImportPath": "k8s.io/heapster/metrics/apis/metrics/v1alpha1",
 			"Comment": "v1.1.0-beta2",
 			"Rev": "9cb18ac0ceb193eb530a1fe339355c94ea454d85"
+		},
+		{
+			"ImportPath": "github.com/google/cadvisor/vendor/github.com/golang/glog",
```
Comment (timstclair) [inline Godeps/Godeps.json]: Why are the vendor dependencies getting pulled in?

Comment (vishh) [inline Godeps/Godeps.json]: I'm not sure why godep is doing that. I'm trying to figure out why.

**Comment by timstclair** [comment]
@Random-Liu

**Comment by Random-Liu** [comment]
@vishh If you run godep save twice, the dlopen will be in.
Seems similar with https://github.com/kubernetes/node-problem-detector/issues/7

**Comment by vishh** [comment]
@Random-Liu that seems plausible. Maybe godep is not detecting recursive dependencies in the first run.

**Comment by vishh** [comment]
However, the `godep-verify.sh` script is going to run godep save just once. I wonder if the script will be happy with running godep save twice.

**Comment by vishh** [comment]
I ran `godep-save.sh` again and noticed no changes on top of my PR.

**Comment by Random-Liu** [comment]
@vishh It's wired. It works for me, maybe because I changed runc to the same version align with docker v1.11.2? https://github.com/docker/docker/tree/v1.11.2/vendor/src/github.com/opencontainers/runc

**Comment by Random-Liu** [comment]
@vishh However, I just run `godep-verify.sh` and it also failed and told me to remove `dlopen` from Godeps...
```
Your Godeps.json is different:
--- /usr/local/google/home/lantaol/workspace/src/k8s.io/kubernetes/Godeps/Godeps.json	2016-07-19 18:11:32.692929121 -0700
+++ /tmp/gopath.ppLOVU/src/k8s.io/kubernetes/Godeps/Godeps.json	2016-07-19 20:40:34.219648761 -0700
@@ -497,11 +497,6 @@
 			"Rev": "8b320e7c550067b1dfb37bd1682e8067023e0751"
 		},
 		{
-			"ImportPath": "github.com/coreos/go-etcd/etcd",
-			"Comment": "v2.0.0-38-g003851b",
-			"Rev": "003851be7bb0694fe3cc457a49529a19388ee7cf"
-		},
-		{
 			"ImportPath": "github.com/coreos/go-oidc/http",
 			"Rev": "5cf2aa52da8c574d3aa4458f471ad6ae2240fe6b"
 		},
@@ -560,11 +555,6 @@
 			"Comment": "v2",
 			"Rev": "7f080b6c11ac2d2347c3cd7521e810207ea1a041"
 		},
-		{
-			"ImportPath": "github.com/coreos/pkg/dlopen",
-			"Comment": "v2",
-			"Rev": "7f080b6c11ac2d2347c3cd7521e810207ea1a041"
-		},
 		{
 			"ImportPath": "github.com/coreos/pkg/health",
 			"Comment": "v2",
```

This is really stupid... :'(

**Comment by Random-Liu** [comment]
@vishh BTW, I don't know why my local godep keeps trying to add `go-etcd/etcd`...

I've to go home now. Will have another try after I get home and finish dinner. :)

**Comment by vishh** [comment]
@Random-Liu I  have updated the presubmit script to run godep twice. I hope this works for now.

**Comment by vishh** [comment]
@k8s-bot test this github issue #IGNORE

**Comment by Random-Liu** [comment]
@freeformz We encountered the same issue again. We have to run godep save twice to include the dependency `github.com/coreos/pkg/dlopen`.

Any suggestions are appreciate. :'(

**Comment by vishh** [comment]
For some reason the ginkgo dependencies were also updated as part of this PR and resulted in an older version of ginkgo as a dep. I tried to fix it now.

**Comment by vishh** [comment]
@timstclair @Random-Liu This PR is finally ready to be merged. PTAL

File: hack/godep-save.sh [hunk 2]
```diff
@@ -31,4 +31,5 @@ REQUIRED_BINS=(
 
 pushd "${KUBE_ROOT}" > /dev/null
   GO15VENDOREXPERIMENT=1 ${GODEP} save "${REQUIRED_BINS[@]}"
+  GO15VENDOREXPERIMENT=1 ${GODEP} save "${REQUIRED_BINS[@]}"
```
Comment (Random-Liu) [inline hack/godep-save.sh]: Is this for temporary walkaround? If so, maybe we need some comments here to explain or add a TODO.

**Comment by Random-Liu** [comment]
Fixes https://github.com/kubernetes/kubernetes/issues/28997.

File: hack/godep-save.sh (hunk 2 above)
Comment (vishh) [inline hack/godep-save.sh]: Done

**Comment by Random-Liu** [comment]
LGTM. :)

**Comment by apelisse** [comment]
This seem to have broken the build.

**Comment by vishh** [comment]
I wonder why this PR was even merged in the first place? Is cross
compilation not required to succeed before merging?
cc @lavalamp

**Comment by lavalamp** [comment]
it was too slow-- we need a separate presubmit to test that.

**Comment by vishh** [comment]
Just the cross-compilation part or `make release`? Can we try
cross-compiling just `cmd/kubectl` as part of presubmit?

**Comment by lavalamp** [comment]
cross-compile specifically. Can't make existing presubmits slower than they already are. Single-platform `make quick-release` is already run.

**Pull request merged**